src/deps_deploy/maven_settings.clj

```diff
@@ -0,0 +1,135 @@
+
+;; Adapted from https://github.com/jelmerk/maven-settings-decoder/blob/master/src/main/java/org/github/jelmerk/maven/settings/Decoder.java
+
+(defn decode-password
+  "Decodes a password (got from settings.xml)."
+  ^String [^String encoded-password ^String key]
+  (-> (DefaultPlexusCipher.)
+      (.decryptDecorated encoded-password key)))
+
+(defn read-settings-security
+  "Reads settings-security.xml file into an ^SettingsSecurity object.
+   Defaults to $HOME/.m2/settings-security.xml ."
+  (^SettingsSecurity []
+   (read-settings-security nil))
+  (^SettingsSecurity [settings-security-path]
+   (let [settings-security-path (or settings-security-path default-settings-security-path)]
+     (SecUtil/read (.getAbsolutePath (io/as-file settings-security-path)) true))))
+
+(defn read-settings
+  "Reads settings.xml file into a ^Settings object.
+   Defaults to $HOME/.m2/settings.xml."
+  (^Settings []
+   (read-settings nil))
+  (^Settings [settings-path]
+   (let [settings-path (or settings-path default-settings-path)
+         sr (SettingsXpp3Reader.)]
+     (with-open [rdr (io/reader (io/as-file settings-path))]
+       (.read sr rdr)))))
+
+(defn decode-server-password
+  "Decodes the server passowr, given a ^Server and plain master password."
+  ^String [^String plain-master-pw ^Server server]
+  (decode-password (.getPassword server) plain-master-pw))
+
+(defn server-credentials
+  "Returns a map from server id -> server settings including decoded password and ^Server instance."
+  [plain-master-pw server]
+  (let [id (.getId server)
+        username (.getUsername server)
+        decoded-pw (decode-password (.getPassword server) plain-master-pw)]
+    {id {:id id
+         :username username
+         :password decoded-pw
+         :_server server}}))
+
+(defn servers-with-passwords
+  "Decodes the passwords from servers.
+   Returns a map from server id -> server settings (including credentials)."
+  ([]
+   (servers-with-passwords (read-settings) (read-settings-security)))
+  ([^Settings settings
+    ^SettingsSecurity settings-security]
+   (let [encoded-master-pw (.getMaster settings-security)
+         plain-master-pw (decode-password encoded-master-pw DefaultSecDispatcher/SYSTEM_PROPERTY_SEC_LOCATION)
+         servers (.getServers settings)]
+     (into {} (map (partial server-credentials plain-master-pw) servers)))))
+
+(defn active-profiles
+  "Map of active profile name to ^org.apache.maven.settings.Profile instance."
+  ([]
+   (active-profiles (read-settings)))
+  ([^Settings settings]
+   (let [active-profiles (.getActiveProfiles settings)
+         profiles-map (.getProfilesAsMap settings)]
+     (select-keys profiles-map active-profiles))))
+
+(defn active-repositories
+  "Returns a list of active repositories from settings.xml active profiles.
+   Does not include crededentials."
+  ([]
+   (active-repositories (read-settings)))
+  ([^Settings settings]
+   (let [active-profiles (active-profiles settings)
```

Comment:
Since you're not using `settings` anywhere in this fn other than to get `active-profiles` you might as well just pass `active-profiles` as a param.

Reply:
I see your point but I'm not sure about this. 
I like having the functions use only Settings. 
User has to read the settings and pass it to functions to get stuff from there.